pandas/core/reshape/tile.py

```diff
@@ -364,6 +365,8 @@ def _bins_to_cuts(x, bins, right=True, labels=None,
             result = result.astype(np.float64)
             np.putmask(result, na_mask, np.nan)
 
+    bins = _convert_bin_to_datelike_type(bins, dtype)
```

Comment:
is there a way to make this logic flow better, e.g. ``_convert_bins_to_numeric_type`` and ``_convert_bin_to_datelike_type`` are disjointed (e.g. they are separate), maybe combine? I just find that we are doing conversions in 2 places here

Reply:
Well they both serve different purposes; 
`_convert_bins_to_numeric_type ` is used at the start to prep the bins (datelike data -> numeric data) before the main array is cut.

 `_convert_bin_to_datelike_type` is used at the end to "stylize" the bins (numeric  data -> datelike data) after the array is cut. 

Plus, I like the explicit naming of these two processes. Just my 2c.

Reply:
ok that's fine then.